### PR TITLE
[Expression] Add resolve function for resolving TimexProperty values to string values

### PR DESCRIPTION
**Pull request by cosmicshuai**
fixes: #4425 
resolve can resolve the timex values to string values.
for example:
resolve('2012-12-30T14') -> "2012-12-30 14:00:00"
resolve('T14') -> "14:00:00"
resolve('2012-12-30') -> "2012-12-30"

for the senario in the reported issue. 
```
#ResolveDateTime
- ${formatDateTime(concat(resolve("2009-01-23"), "T", resolve("T14")))}
```

it will return `"2009-01-23T14:00:00:00.000Z"`

### CODE REVIEW

**Comment by boydc2014** [comment]
we should backward support the notion without prefix like "getNextViableTime", for at least one release i guess.

File: libraries/AdaptiveExpressions/ExpressionType.cs [hunk 1]
```diff
@@ -114,17 +114,18 @@ public static class ExpressionType
...
+        public const string IsDuration = "timex.isDuration";
+        public const string IsDate = "timex.isDate";
+        public const string IsTimeRange = "timex.isTimeRange";
+        public const string IsDateRange = "timex.isDateRange";
+        public const string IsPresent = "timex.isPresent";
+        public const string GetNextViableDate = "timex.getNextViableDate";
+        public const string GetPreviousViableDate = "timex.getPreviousViableDate";
+        public const string GetNextViableTime = "timex.getNextViableTime";
+        public const string GetPreviousViableTime = "timex.getPreviousViableTime";
+        public const string TimexResolve = "timex.resolve";
```
Comment (chrimc62) [inline libraries/AdaptiveExpressions/ExpressionType.cs:128]: Not sure how I feel about putting these all in a namespace.  We have not done this anywhere else.  Was @vishwacsena involved?

**Review by chrimc62 — CHANGES_REQUESTED** [review]
:clock1:

File: libraries/AdaptiveExpressions/ExpressionType.cs (hunk 1 above)
Comment (boydc2014) [reply libraries/AdaptiveExpressions/ExpressionType.cs:128]: yes, this is based on @vishwacsena's suggestion, i'm OK with both, with or without namespace.  I feel like at least namespace help distinguish "timex.resolve"

**Comment by GeoffCoxMSFT** [comment]
@EricDahlvang - Could you please review this PR as it is a customer reported issue.

File: libraries/AdaptiveExpressions/ExpressionType.cs (hunk 1 above)
Comment (chrimc62) [reply libraries/AdaptiveExpressions/ExpressionType.cs:128]: I talked to Vishwac and we agreed that we should remove all of the timex prefixes.  We don't do this for any other data types, like string or object so we should not do it here.

---
In reply to: [486108530](https://github.com/microsoft/botbuilder-dotnet/pull/4573#discussion_r486108530) [](ancestors = 486108530)

Comment (boydc2014) [reply libraries/AdaptiveExpressions/ExpressionType.cs:128]: ok, let's do it

**Comment by EricDahlvang** [comment]
@chrimc62 My level of familiarity with this library is minimal.  Does this look good to you?

File: libraries/AdaptiveExpressions/ExpressionType.cs (hunk 1 above)
Comment (chrimc62) [inline libraries/AdaptiveExpressions/ExpressionType.cs:128]: Remove timex. prefix.

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs [hunk 2]
```diff
@@ -213,6 +214,20 @@ public static class ExpressionFunctions
...
+            lookup.Add("isDate", lookup[ExpressionType.IsDate]);
+            lookup.Add("isTimeRange", lookup[ExpressionType.IsTimeRange]);
+            lookup.Add("isDateRange", lookup[ExpressionType.IsDateRange]);
+            lookup.Add("isPresent", lookup[ExpressionType.IsPresent]);
+            lookup.Add("getNextViableDate", lookup[ExpressionType.GetNextViableDate]);
+            lookup.Add("getPreviousViableDate", lookup[ExpressionType.GetPreviousViableDate]);
+            lookup.Add("getNextViableTime", lookup[ExpressionType.GetNextViableTime]);
+            lookup.Add("getPreviousViableTime", lookup[ExpressionType.GetPreviousViableTime]);
+            lookup.Add("resolve", lookup[ExpressionType.TimexResolve]);
+
```
Comment (chrimc62) [inline libraries/AdaptiveExpressions/ExpressionFunctions.cs:230]: This is not necessary once you remove timex prefix.

File: libraries/AdaptiveExpressions/ExpressionType.cs (hunk 1 above)
Comment (cosmicshuai) [reply libraries/AdaptiveExpressions/ExpressionType.cs]: removed

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs (hunk 2 above)
Comment (cosmicshuai) [reply libraries/AdaptiveExpressions/ExpressionFunctions.cs]: removed alias here and related tests

**Comment by cosmicshuai** [comment]
@chrimc62 
Hi Chris, I removed the prefix of all timex functions. Could you take a look? Thanks a lot.

**Review by chrimc62 — APPROVED** [review]

**Pull request merged**